### DISCUSSION TITLE
Discussion for overhaul for UPS power and related sci-fi items

### DISCUSSION THREAD

**Discussion by chaosvolt** [cataclysmbn/Cataclysm-BN#2053] [General]
# Is your feature request related to a problem? Please describe.

Per some discussion in recent PRs and on the BN discord, I decided to finally open an issue for discussing and tracking what we plan to do with UPS-powered items, how they should be balanced relative to battery-powered items, and how they interact with other mechanics like the grid system.

Current handling of how the UPS mechanics are used leaves several questions open, both in terms of lore and game balance, both of which are largely due to inherited half-written lore and half-finished rebalancing from DDA. Being able to outline all problems here and outline a plan for what to do will make those PRs less of a tangled mess to sort out.

Lore-wise, the following questions are evident:
1. What is the UPS actually supposed to be? It definitely is not anything like what the IRL term tends to describe (a homemade example: https://www.youtube.com/watch?v=1q4dUt1yK0g). Instead they seem to be a power bank with some sci-fi applications, but it also strongly implies possessing an inverter built in as well. This leads directly to the second question...
2. In what contexts in-universe would you expect a UPS to see use? Is it civilian tech, military, experimental superscience, or all three? The UPS seems be used mainly as a military and science thing, while being rare yet still accessible to civilians, which fits neither a supersoldier power armor fission core thing (rare and non-civilian) nor a real-world-inspired power bank, inverter, what-have-you (power banks and inverters both being common and civilian, with only a portable power bank beefy enough to run an inverter potentially edging into scifi territory). This also offers no meaningful distinction between the standard UPS and the advanced one, which seems in practice to be just the UPS but new and rare. This likewise leads to...
3. What is the functional difference between a standard UPS and an advanced UPS? What roles do each serve in-universe? And what makes an advanced UPS more efficient when fed power cells, as compared to a standard UPS using batteries that are no more or less energetic than the batteries in a plutonium battery, itself made out of plutonium cells? And finally, this leads to:
4. What is the practical difference between the old plutonium cell and the new plutonium battery, lore-wise? In practice, the answer seems to be "half-finished attempt to depreciate minireactors and reflavor plutonium batteries as ayy lmao nanite batteries" which we inherited from DDA.

In gameplay terms, the following issues are raised:
1. Which items should be powered by batteries, and which should be powered by UPS? The correct handling leans to somewhat support the idea that UPS is for both USB-powered items with integral batteries, with a limited application for items that should run on AC power. For the latter, many items are also things that would realistically run on AC power but don't for gameplay reasons.
2. How commonplace is the UPS and advanced UPS, and where should they spawn? As mentioned above, their spawn usage currently strikes me as haphazard and not well thought out. It's civilian but also military, mil-surp and superscience but also commercial, and it only further contributes to the advance UPS not really having a distinct niche relative to the standard variant. It also means that UPS power is effectively a SORTA nerf to items given it, because UPS is inconsistent and unreliable to find due to its unclear logic for where it's meant to exist. But the instant you get a UPS it ceases to be a nerf and instead becomes no less of a problem to keep running than a standard rechargeable battery. The only functional loss is the ability to get lucky and collect a good stash of disposable batteries, which is as much a matter of good RNG as getting that UPS is in the first place, just with the overall likelihood running higher for disposable batteries (meaning you still can blindly stumble into a UPS on day one just as much as you can go forever without finding a charged battery in the size you need, especially if it isn't in the light battery class).
3. How should UPS power interact with grid power? As implied in point one, the bulk of items that PROBABLY should be grid-powered are things that use either batteries or UPS. If we make more items possible to plug in (a suggestion that has come up), it will make converting items into 

# Describe the solution you'd like

There are multiple steps to this, multiple possible ways to go about it too, but a rough outline of things that could be done:
1. Improve and property define what the UPS is, what it does, and where it spawns. This will make it easier to determine what items should be UPS powered, depending on how common the UPS becomes. My proposal is that the standard UPS should lean towards being a hard sci-fi counterpart to a power bank, with enough oomph to justify an inverter being integral to it. This means it would logically be a potential find wherever you might expect portable backup power to be a concern. It'd be associated with consumer electronics, outdoors supplies, automotive and RV contexts, etc. Military and superscience use would instead favor the advanced UPS.
2. Make it feasible to run any items we make UPS-powered off grid power, leaning fully into the idea that anything that should be AC-powered should have the option of being plugged into a working grid. While point one makes UPS power less of an annoying restriction, this would make a grid a viable alternative for any tools that could be plugged in. The idea of a "socket" construction to simplify this came up in the discord and may be a good improvement if this can be coded, as that'd make it less clunky to use than having to slap down a power convertor for every item you want running on your grid.
3. Hash out what items should be UPS-powered by default, and which should stay battery-powered. A good baseline criteria would be to start with any items that would likely have an integral lithium-ion battery. For example cell phones, cameras, and laptops would logically stay battery-powered under this definition, while smartphones, professional cameras, and E-tablet PCs would likely be converted to UPS. Second phase of this would be to set any AC-powered items to be
4. Add a sane way to convert UPS-only items into battery powered. The current way would require adding an entirely new item with attendant recipe. We can add a toolmod to convert battery items to run off UPS, so maybe a craftable toolmod for converting AC/UPS items to run off batteries would be a good option here, per discussion in the discord.
5. Properly define plutonium cells vs. plutonium batteries.  My proposal: plutonium cells should be reflavored to actually make use of both the nano-compounds in it (from modern DDA's ayy lmao lore) AND the plutonium in it (from old), plutonium-cell-powered items are thus implied to use a hybrid of sci-fi power storage supplemented with either fission (soft sci-fi) or alphavolatics (hard sci-fi) to increase efficiency. Plutonium batteries meanwhile involve a process to adapt the cells to fit common battery standards, losing their ability to make effective use of the plutonium but retaining the potential for very high power density.

# Describe alternatives you've considered

Not sure what else comes to mind, alternatives will be added here per proposals or other ideas that come up.

A good supplement to point five: make the plutonium batteries more accessible to find or craft (at least buffed to being a general superscience contents instead of basically only existing in a single finale variant), but IF the crafting recipes are made accessible, add additional components to balance them accordingly. If nanomaterial is made more accessible, this would be a good expansion of the material's uses to being not just nanofab fodder (in the vein of it also being useful for nano-serum in Cataclysm++ mod).

Speaking of Cata++, making the plutonium batteries themselves spawn in more areas could be ported from said mod (done because it uses the plutonium batteries for certain items in the mod), with some fleshing out.

# Additional context

**Comment by chaosvolt** [comment]
Suggestion that came up in the discord: some alternative way to charge a UPS, like gasoline fuel cell CBM but for item batteries, would be a useful expansion to this. Comparison can be made to portable generators, but only generator item that exists is a rare single-tile vehicle with the game's only real occurrence of the largest alternator in vanilla.

Side note: Check if said vehicle is deployable, make it foldable and deployable if it's not not, and look into sanity-checking its spawns. It'd pair very well with the grid system too.

**Comment by mkikt4743** [comment]
Ideally, power sources that are easily nabbable, like chargers in houses, and generally power systems in houses to support sich, would come aswell, or before, big conversion to UPS, since many buildings have lamps but no power.

**Comment by chaosvolt** [comment]
An easier ability to just plug items in would be great for supplementing this, yeah.

**Comment by Coolthulhu** [comment]
I'd like to keep UPS socket and power grid socket separate. It's just that power grid socket doesn't exist yet.

I see it like this:
* Regular batteries for portable things with low power draw, up to about hotplate
* "UPS battery" for things that can be recharged, but can't have batteries swapped. Mostly electronics with vendor locks - smartphones, laptops, "gadgets"
* UPS slot with no own battery for things that should be light, but have high power draw. Weapons, handheld power tools like chainsaw, welders.
* UPS could support power grid slot, but not the other way around

I'm not 100% sure about the UPS vs power grid split, it may be worth merging.

**Comment by chaosvolt** [comment]
Not sure what you mean about a split between UPS sockets and power grid sockets? My thinking was basically that most items that should be plugged into an electrical output would probably also work if used from a UPS. Which, as mentioned, would benefit from some code support.

Logically there would be things that can only draw from a UPS and can't use the power grid, like weapons and power armor. I assume that's what you mean but UPS supporting power grid slots but not vice-versa?
